File: csharp/gabaritos/gabarito20.cs
using System;
using System.Collections.Generic;

class Program
{
    static int ContarAninhado(List<object> lst, int alvo)
    {
        int count = 0;
        foreach (object item in lst)
        {
            if (item is List<object> sublist)
            {
                count += ContarAninhado(sublist, alvo);
            }
            else if (item is int num && num == alvo)
            {
                count++;
            }
        }
        return count;
    }

    static void Main()
    {
        List<object> lista = new List<object> { new List<int> { 1, 2 }, new List<int> { 2, 3 }, 2 };
        int alvo = 2;
        Console.WriteLine($"Contagem de {alvo}: {ContarAninhado(lista, alvo)}");
    }
}

File: cpp/gabaritos/gabarito20.cpp
#include <iostream>
#include <vector>

int contar_aninhado(const std::vector<std::vector<int>>& lst, int alvo, size_t index = 0) {
    if (index == lst.size()) return 0;
    int count = 0;
    for (int num : lst[index]) {
        if (num == alvo) ++count;
    }
    return count + contar_aninhado(lst, alvo, index + 1);
}

int main() {
    std::vector<std::vector<int>> lista = {{1, 2}, {2, 3}, {2}};
    int alvo = 2;
    std::cout << "Contagem de " << alvo << ": " << contar_aninhado(lista, alvo) << std::endl;
    return 0;
}